File: backend/src/main/java/fun/asem/biograph/webapp/domain/Event.java
package fun.asem.biograph.webapp.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.LazyCollection;
import org.hibernate.annotations.LazyCollectionOption;

import javax.persistence.*;
import java.time.Instant;
import java.util.List;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "events")
@Data
@Builder
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long eventId;
    private String name;
    private String description;
    private Instant startDatetime;
    private Instant endDatetime;
    private Instant creationTime;
    private Instant lastModifiedTime;
    @LazyCollection(LazyCollectionOption.FALSE)
    @OneToMany(mappedBy = "event", cascade = CascadeType.ALL)
    private List<Parameter> parameters;
    @LazyCollection(LazyCollectionOption.FALSE)
    @OneToMany(cascade = CascadeType.ALL, mappedBy = "event")
    private List<Attachment> attachments;
    @ManyToMany
    @JoinTable(
            name = "event_tags",
            joinColumns = {@JoinColumn(name = "event_id")},
            inverseJoinColumns = {@JoinColumn(name = "tag_id")}
    )
    private List<Tag> tags;
    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
    @OneToMany(cascade = CascadeType.ALL, mappedBy = "event", fetch = FetchType.LAZY)
    private List<Grant> grants;
}

File: biograph-frontend/src/app/models/Event.ts
import {Tag} from './Tag';
import {Category} from './Category';
import {Parameter} from './Parameter';
import {Attachment} from './Attachment';

export interface Event {
  eventId: number;
  name: string;
  description: string;
  startDatetime: string;
  endDatetime: string;
  creationTime: string;
  lastModifiedTime: string;
  parameters: Parameter[];
  attachments: Attachment[];
  tags: Tag[];
  category: Category;
}

export function getStubEmptyEvent(): Event {
  return {
    eventId: null,
    name: '',
    description: '',
    startDatetime: '',
    endDatetime: '',
    creationTime: '',
    lastModifiedTime: '',
    parameters: [],
    attachments: [],
    tags: [],
    category: null,
  };
}
